tests/test_matrix.py

```diff
@@ -112,3 +114,64 @@ def test_decode():
     expected = init_matrix(2, Polynomial([Monomial('m', [(0, 1)])]))
 
     assert decoded == expected
+
+
+def test_matrix_equals():
+    p1 = Polynomial([Monomial('m', [(0, 1), (1, 1)])])
+    p2 = Polynomial([Monomial('m', [(0, 1), (1, 1)])])
+    p3 = Polynomial([Monomial('m', [(0, 0)])])
+    p4 = Polynomial([Monomial('m', [(0, 0)])])
+    p5 = Polynomial([Monomial('m', [(1, 1), (2, 2)])])
+    p6 = Polynomial([Monomial('m', [(1, 1), (2, 2)])])
+
+    m1 = [[o, p1, o], [p3, o, o], [o, o, p5]]
+    m2 = [[o, p2, o], [p4, o, o], [o, o, p6]]
+
+    assert equals(m1, m2) is True
+
+
+def test_matrix_not_equals():
+    p1 = Polynomial([Monomial('m', [(0, 1)])])
+    p2 = Polynomial([Monomial('m', [(1, 1)])])
+
+    m1 = [[o, o, o], [o, o, o], [o, o, p1]]
+    m2 = [[o, o, o], [o, o, o], [o, o, p2]]
+
+    assert equals(m1, m2) is False
+
+
+def test_matrix_size_not_equals():
+    m1 = [[o, o, o], [o, o, o], [o, o, o]]
+    m2 = [[o, o], [o, o]]
+
+    assert equals(m1, m2) is False
+
+
+def test_fixpoint():
```

Comment:
Great. I can try to come up with more complex examples at some point.

Reply:
Yes, I would like to have a more advanced case to test. If you spot one that changes when doing things by hand make a note of it for me please.